src/network/tc/tc-util.c

```diff
@@ -92,3 +93,25 @@ int tc_fill_ratespec_and_table(struct tc_ratespec *rate, uint32_t *rtab, uint32_
         rate->linklayer = TC_LINKLAYER_ETHERNET;
         return 0;
 }
+
+int tc_parse_qdisc_handle(const char *s, uint32_t *h) {
+        uint32_t m;
+        char *p;
+
+        m = TC_H_UNSPEC;
+        if (streq(s, "none")) {
+                *h = m;
+                return 0;
+        }
+
+        m = strtoul(s, &p, 16);
+        if (p == s || m >= (1 << 16))
+                return -EINVAL;
+
+        m <<= 16;
+        if (*p != ':' && *p != 0)
```

Comment:
For qdisc, it is not necessary to add `:` at the end, as the minor number never specified. Let's simply takes major number without colon.

Reply:
updated yes.